src/kubernetes.cc

```diff
@@ -843,6 +903,41 @@ void KubernetesReader::WatchNodes(
   LOG(INFO) << "Watch thread (node) exiting";
 }
 
+void KubernetesReader::ResourceCallback(
+    MetadataUpdater::UpdateCallback callback,
+    const json::Object* resource, Timestamp collected_at, bool is_deleted) const
+    throw(json::Exception) {
+  std::vector<MetadataUpdater::ResourceMetadata> result_vector;
+  result_vector.emplace_back(
+      GetResourceMetadata(resource, collected_at, is_deleted));
+  callback(std::move(result_vector));
+}
+
+void KubernetesReader::WatchResources(
+    const std::string& api_path, const std::string& name,
+    MetadataUpdater::UpdateCallback callback) const {
+  LOG(INFO) << "Watch thread (" << name << ") started";
+
+  try {
+    // TODO: There seems to be a Kubernetes API bug with watch=true.
```

Comment:
Does this TODO belong here? If so, clarify what we should do to address it. Thanks.

Reply:
Good catch. This is a copy paste from a prototype code. Since I don't remember the origin, deleting it.